public/app/AppWrapper.tsx

```diff
@@ -109,33 +108,18 @@ export class AppWrapper extends Component<AppWrapperProps, AppWrapperState> {
                 actions={[]}
                 options={{ enableHistory: true, callbacks: { onSelectAction: commandPaletteActionSelected } }}
               >
-                <Router history={locationService.getHistory()}>
-                  <LocationServiceProvider service={locationService}>
-                    <CompatRouter>
-                      <ModalsContextProvider>
-                        <GlobalStyles />
-                        <div className="grafana-app">
-                          <AppChrome>
-                            <AngularRoot />
-                            <AppNotificationList />
-                            <Stack gap={0} grow={1} direction="column">
-                              {pageBanners.map((Banner, index) => (
-                                <Banner key={index.toString()} />
-                              ))}
-                              {ready && this.renderRoutes()}
-                            </Stack>
-                            {bodyRenderHooks.map((Hook, index) => (
-                              <Hook key={index.toString()} />
-                            ))}
-                          </AppChrome>
-                        </div>
-                        <LiveConnectionWarning />
-                        <ModalRoot />
-                        <PortalContainer />
-                      </ModalsContextProvider>
-                    </CompatRouter>
-                  </LocationServiceProvider>
-                </Router>
+                <GlobalStyles />
+                <SidecarContext.Provider value={sidecarService}>
+                  <div className="grafana-app">
+                    {config.featureToggles.appSidecar ? (
+                      <ExperimentalSplitPaneRouterWrapper {...routerWrapperProps} />
+                    ) : (
+                      <RouterWrapper {...routerWrapperProps} />
+                    )}
+                    <LiveConnectionWarning />
+                    <PortalContainer />
+                  </div>
+                </SidecarContext.Provider>
```

Comment:
ideally this would have exactly the same structure when the toggle is off. 

i get it probably would have been more code and you've tried to simplify a bit etc, so this is fine, but i think this is the part that makes it a little riskier/harder to review 😅

Reply:
I agree that is a bit higher risk, at the same if there is something wrong I would really want to know because it may be a blocker for the future of this. What offsets this in my mind apart from the testing is that if anything is wrong I think it will be noticed very quickly considering it's the upper level app component structure.